test_vectors.py

```diff
@@ -1,9 +1,32 @@
 import unittest
 
-from balloon import balloon, balloon_hash, balloon_m, balloon_m_hash, verify, verify_m
+from balloon import (
+    _balloon,
+    balloon,
+    balloon_hash,
+    balloon_m,
+    balloon_m_hash,
+    verify,
+    verify_m,
+)
 
 
 class TestBalloon(unittest.TestCase):
+    def test_invalid_params(self):
+        test_vectors = [
+            {"args": ("", "", -1, 0, 0), "param": "space_cost"},
+            {"args": ("", "", 0, -1, 0), "param": "time_cost"},
+            {"args": ("", "", 0, 0, -1), "param": "delta"},
+        ]
+
+        for test_vector in test_vectors:
+            with self.assertRaises(ValueError) as context:
+                _balloon(*test_vector["args"])
```

Comment:
Do we really have to use the internal function here?

Reply:
The same code coverage can be achieved by testing `balloon` instead, however, `_balloon` is used by both `balloon` and `balloon_m`, and the validation checks are done in `_balloon` so I thought it would be appropriate to test the internal `_balloon`.

Reply:
Well, good point.
Could you leave a comment here stating that?